File: Head_First_Design_Patterns/src/facade/HomeTheaterTestDrive.java
package facade;

public class HomeTheaterTestDrive {
    public static void main(String[] args){
        Amplifier amplifier = new Amplifier("Top-O-Line Amplifier");
        Tuner tuner = new Tuner("Top-O-Line AM/FM Tuner", amplifier);
        DvdPlayer dvdPlayer = new DvdPlayer("Top-O-Line DVD Player", amplifier);
        CdPlayer cdPlayer = new CdPlayer("Top-O-Line CD Player", amplifier);
        Projector projector = new Projector("Top-O-Line Projector", dvdPlayer);
        TheaterLights lights = new TheaterLights("Theater Ceiling Lights");
        Screen screen = new Screen("Theater Screen");
        PopcornPopper popper = new PopcornPopper("Popcorn Popper");

        HomeTheaterFacade homeTheaterFacade = new HomeTheaterFacade(amplifier, tuner, dvdPlayer, cdPlayer, projector, screen, lights, popper);

        homeTheaterFacade.watchMovie("F***************k");
        homeTheaterFacade.endMovie();

        homeTheaterFacade.listenToCd("Kamikaze");
        homeTheaterFacade.endCd();
    }
}

File: Head_First_Design_Patterns/src/facade/python/HomeTheaterTestDrive.py
# -*- coding: utf-8 -*-
"""
Created on Wed Oct 24 18:10:48 2018

@author: 凯风
"""

from sub import Amplifier,Tuner,DvdPlayer,CdPlayer,PopcornPopper,TheaterLights,Screen,Projector
from HomeTheaterFacade import HomeTheaterFacade

if __name__ == "__main__":
    amplifier = Amplifier("Top-O-Line Amplifier")
    tuner = Tuner("Top-O-Line AM/FM Tuner", amplifier)
    dvdPlayer = DvdPlayer("Top-O-Line DVD Player", amplifier)
    cdPlayer = CdPlayer("Top-O-Line CD Player", amplifier)
    projector = Projector("Top-O-Line Projector", dvdPlayer)
    lights = TheaterLights("Theater Ceiling Lights")
    screen = Screen("Theater Screen")
    popper = PopcornPopper("Popcorn Popper")
    
    homeTheaterFacade = HomeTheaterFacade(amplifier, tuner, dvdPlayer, cdPlayer, projector, screen, lights, popper)
    
    homeTheaterFacade.watchMovie("F***************k")
    homeTheaterFacade.endMovie()
    
    print("\n")
                              
    homeTheaterFacade.listenToCd("Kamikaze")
    homeTheaterFacade.endCd()
